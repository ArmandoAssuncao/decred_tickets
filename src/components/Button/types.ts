export interface IProps {
    m: string
    onPress: () => void
    title: string
}

export interface IState {
}
