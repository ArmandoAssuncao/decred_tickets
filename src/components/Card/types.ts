import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    type: ITicket['status']
    value: number
}

export interface IState {
}
