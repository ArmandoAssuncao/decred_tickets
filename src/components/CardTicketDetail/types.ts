import { ITicket } from 'store/ticket/types'

export interface IProps {
    onPress?: () => void
    ticket: ITicket
}

export interface IState {
}
