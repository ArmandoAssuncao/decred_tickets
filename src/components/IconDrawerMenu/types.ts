export interface IProps {
    color?: string
    onPress?: () => void
    size?: number
}

export interface IState {
}
