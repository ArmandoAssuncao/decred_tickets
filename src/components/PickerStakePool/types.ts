import { ITicketState } from 'store/ticket/types'

export interface IProps {
    value?: ITicketState['stakePool']
    onChange?: (value: ITicketState['stakePool']) => void
}

export interface IState {
}
