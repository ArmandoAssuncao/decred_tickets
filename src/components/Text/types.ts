export interface IProps {
    children?: string
    id?: string
    m: string
}

export interface IState {
}
