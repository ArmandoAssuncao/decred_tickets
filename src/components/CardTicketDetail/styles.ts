import { ViewStyle } from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'

export default EStyleSheet.create({
    container: {
        backgroundColor: '$secondColor',
        padding: '0.8rem',
        margin: '0.2rem',
        flexDirection: 'row',
        borderRadius: 5,
        flex: 1,
    } as ViewStyle,
    image: {
        height: 30,
        width: 30,
    } as ViewStyle,
    imageContainer: {
        alignItems: 'center',
        justifyContent: 'center',
    } as ViewStyle,
    textContainer: {
        flex: 1,
        justifyContent: 'center',
        marginLeft: '0.6rem',
    } as ViewStyle,
    transactionContainer: {
        marginTop: '0.4rem',
    } as ViewStyle,
})
