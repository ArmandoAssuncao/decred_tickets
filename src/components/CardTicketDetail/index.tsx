import React from 'react'
import { Image, View, TouchableOpacity } from 'react-native'
import moment from 'moment'

import { IProps, IState } from './types'
import styles from './styles'
import _ from 'lodash'

import { Text } from 'components'

export default class CardTicketDetail extends React.Component<IProps, IState> {
    public static defaultProps: Partial<IProps> = {
    }

    render(): JSX.Element {
        const { onPress, ticket } = this.props
        let img = null
        switch (ticket.status) {
            case 'revoked':
                img = require('@assets/imgs/ticket-revoked.png')
                break
            case 'voted':
                img = require('@assets/imgs/ticket-voted.png')
                break
            case 'expired':
                img = require('@assets/imgs/ticket-expired.png')
                break
            case 'missed':
                img = require('@assets/imgs/ticket-missed.png')
                break
            case 'unmined':
                img = require('@assets/imgs/ticket-unmined.png')
                break
            case 'immature':
                img = require('@assets/imgs/ticket-immature.png')
                break
            case 'live':
                img = require('@assets/imgs/ticket-live.png')
                break
            default:
                break
        }

        return (
            <TouchableOpacity onPress={onPress} style={styles.container}>
                <View style={styles.imageContainer}>
                    <Image
                        style={styles.image}
                        source={img}
                    />
                </View>
                {
                    _.isObject(ticket)
                    ? (
                        <View style={styles.textContainer}>
                            <Text m={'--'} size="xl" type="bold">{ticket.ticketPrice.toString()}</Text>
                            <Text m={'--'} size="md">{moment(ticket.buyTime).format('LLL')}</Text>
                            <View style={styles.transactionContainer}>
                                <Text
                                    m={'Transaction'}
                                    id={'components.cardTicketDetails.transaction'}
                                    size="md"
                                />
                                <Text m={'--'} size="md">{ticket.buyTxId}</Text>
                            </View>
                        </View>
                    ) : null
                }
            </TouchableOpacity>
        )
    }
}
