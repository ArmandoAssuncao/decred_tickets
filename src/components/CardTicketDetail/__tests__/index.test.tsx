import React from 'react'
import 'react-native'
import { shallow } from 'enzyme'

import CardTicketDetail from '../index'
import { ITicket } from 'store/ticket/types'

describe('CardTicketDetail', () => {
    const ticket: ITicket = {
        buyTime: 0,
        buyTxId: 'buyTxId',
        feeStakePool: 0,
        returnTime: 0,
        returnTotal: 0,
        returnTxId: 'returnTxId',
        reward: 0,
        status: 'live',
        ticketFee: 0,
        ticketPrice: 0,
        totalInvestment: 0,
    }
    const wrapper = shallow(<CardTicketDetail onPress={jest.fn()} ticket={ticket} />)

    it('match snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('has 1 TouchableOpacity component', () => {
        expect(wrapper.find('TouchableOpacity')).toHaveLength(1)
    })

    it('has 4 Component component', () => {
        expect(wrapper.find('Component')).toHaveLength(4)
    })

    it('has 4 Text component', () => {
        expect(wrapper.find('Text')).toHaveLength(4)
    })

    it('has 1 Image component', () => {
        expect(wrapper.find('Image')).toHaveLength(1)
    })

    describe('actions', () => {
        it('call onPress when pressed', () => {
            wrapper.props().onPress()
            expect(wrapper.prop('onPress')).toBeCalled()
        })
    })
})
