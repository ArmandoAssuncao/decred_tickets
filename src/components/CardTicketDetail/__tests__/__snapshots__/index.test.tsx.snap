// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTicketDetail match snapshot 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[MockFunction]}
  style={
    Object {
      "backgroundColor": "#F3F6F6",
      "borderRadius": 5,
      "flex": 1,
      "flexDirection": "row",
      "margin": 3.2,
      "padding": 12.8,
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "height": 30,
          "width": 30,
        }
      }
    />
  </Component>
  <Component
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "marginLeft": 9.6,
      }
    }
  >
    <Text
      m="--"
      size="xl"
      type="bold"
    >
      0
    </Text>
    <Text
      m="--"
      size="md"
      type="normal"
    >
      December 31, 1969 9:00 PM
    </Text>
    <Component
      style={
        Object {
          "marginTop": 6.4,
        }
      }
    >
      <Text
        id="components.cardTicketDetails.transaction"
        m="Transaction"
        size="md"
        type="normal"
      />
      <Text
        m="--"
        size="md"
        type="normal"
      >
        buyTxId
      </Text>
    </Component>
  </Component>
</TouchableOpacity>
`;
